altanalyze3/components/merging/main.py

```diff
@@ -0,0 +1,171 @@
+from operator import truediv
+import pandas as pd
+import os
+
+#path for gene_model and junction files
+
+junction_dict = {}
+exon_dict = {}
+gene_dict = {}
+
+def sampleBEDFileAnnotation(gene_model_all, junction_dir, intron_dir):
+    exon_dict, gene_dict,junction_dict = importJunctionInfo(gene_model_all)
+    importJunctionFromBED(junction_dir, junction_dict)
+
+
+def importJunctionInfo(gene_model_all):
+    '''
+    This method will take gene_model as input and create three dictionaries
+    1. junction_dict
+    2. exon_dict
+    3. gene_dict
+    '''
+    
+
+    gene_model_df = pd.read_csv(gene_model_all,sep='\t',header=None, names=[
+            "gene_id", "chr", "strand","exon_region_id", "start", "stop", "exon_annotations"])    
+    #junction_dict = gene_model_df.set_index('chr','start').to_dict()['gene_id']
+    for idx,row in gene_model_df.iterrows():
+        junction_dict[(row.chr,row.start)] = row.gene_id + ':' + row.exon_region_id
+        junction_dict[(row.chr,row.stop)] = row.gene_id + ':' + row.exon_region_id
+        exon_annotation = get_exon_annotation(row.chr,row.start,row.stop,row.exon_region_id)
+        if row.gene_id in exon_dict.keys():
+            exon_dict[row.gene_id].append(exon_annotation)
+            gene_dict[row.gene_id].append(row.start)
+            gene_dict[row.gene_id].append(row.stop)
+        else:
+            exon_dict[row.gene_id] = [exon_annotation]
+            gene_dict[row.gene_id] = [row.start,row.stop]
+    # for gene in gene_dict:
+    #     gene_dict[gene].sort()
+    #print(junction_dict)
+    return exon_dict, gene_dict, junction_dict
+
+   
+def get_exon_annotation(chr, start,stop,exonid):
+    exon_start = chr + ':' + str(start) 
+    exon_stop = chr + ':' + str(stop)
+    exon_id =  exonid + '.' + str(chr)
+    return {'exon_start':exon_start, 'exon_stop': exon_stop,'exon_id':exon_id}
+
+
+def importJunctionFromBED(junction_dir, junction_dict):
+    '''
+    This function imports junction data from BAM derived exon junction text files
+    '''
+    junction_coordinate_BAM_dict = {}
+
+    for junctionfilename in os.listdir(junction_dir):
+        junctionfile = os.path.join(junction_dir, junctionfilename)
+        junction_coordinate_BAM_dict = initialBAMJunctionImport(junctionfile)
```

Comment:
This seems to include information only from the last junctionfile

Reply:
I have updated this code in main_new.py and not using this anymore.